### ISSUE TITLE
Lint/Syntax false negative

### ISSUE THREAD

**Issue by Tolsto** [bbatsov/rubocop#5748]
```
def foo(arg1:)
  puts arg1
  yield
end

foo arg1: {hello: 'world'} { puts 'bar' }
```

Running the above code with Ruby 2.5.1 raises: 

```
syntax error, unexpected '{', expecting end-of-input
foo arg1: {hello: 'world'} { puts 'bar' }
```

However, Lint/Syntax does not report a violation. 

--------

## Expected behavior

Lint/Syntax should report a violation

## Actual behavior

Lint/Syntax does not report a violation

## Steps to reproduce the problem

Run/Inspect the example code with Ruby 2.5.1

## RuboCop version

```
$ rubocop -V
0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by mikegee** [comment]
I confirmed the report. I suspect that this issue belongs on the parser gem since it thinks the example is valid syntax:

```
parser master % ./bin/ruby-parse ../rubocop/test/test.rb
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.0-compliant syntax, but you are running 2.5.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
(begin
  (def :foo
    (args
      (kwarg :arg1))
    (begin
      (send nil :puts
        (lvar :arg1))
      (yield)))
  (block
    (send nil :foo
      (hash
        (pair
          (sym :arg1)
          (hash
            (pair
              (sym :hello)
              (str "world"))))))
    (args)
    (send nil :puts
      (str "bar"))))
```

(Note the warning about "2.5.0-compliant syntax", which implies Parser doesn't support 2.5.1 yet.)

**Comment by Drenmi** [comment]
@Tolsto: Please open an issue with [parser](https://github.com/whitequark/parser). 🙂

**Comment by Tolsto** [comment]
@Drenmi Parser 2.5.1.0 has been released which fixes https://github.com/whitequark/parser/issues/495
